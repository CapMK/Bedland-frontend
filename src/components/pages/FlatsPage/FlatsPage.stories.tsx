import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import ThemeProvider from 'theme/ThemeContext';
import GlobalStyle from 'theme/globalStyles';

import { Provider } from 'react-redux';
import { reduxStore } from 'store/store';
import FlatsPage from './FlatsPage';

const meta = {
  title: 'Pages/FlatsPage',
  component: FlatsPage,
  tags: ['autodocs'],
  parameters: {
    backgrounds: { default: 'light' },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        <Provider store={reduxStore}>
          <BrowserRouter>{Story()}</BrowserRouter>
        </Provider>
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof FlatsPage>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
