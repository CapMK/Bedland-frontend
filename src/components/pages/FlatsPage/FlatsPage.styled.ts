import styled from 'styled-components';

export const SearchInputContainer = styled.div`
  display: flex;
  justify-content: flex-end;
  margin: 0.5rem 0;
`;

export const FlatsPageContent = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;
