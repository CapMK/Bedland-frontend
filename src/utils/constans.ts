export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
