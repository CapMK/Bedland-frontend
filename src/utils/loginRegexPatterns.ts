import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
