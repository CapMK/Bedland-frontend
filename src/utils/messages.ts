const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
