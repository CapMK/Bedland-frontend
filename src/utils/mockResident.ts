const mockResidentList = [
  {
    id: 1,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'iSutherland',
    password: 'IsABEl',
    name: 'Isabel',
    lastName: 'Sutherland',
    email: 'isabel.sutherland@email.com',
    phoneNumber: '123456789',
    flatId: 1,
    owner: true,
  },
  {
    id: 2,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'clWeiss',
    password: 'ClWeiss',
    name: 'Claude',
    lastName: 'Weiss',
    email: 'Claude.Weiss@email.com',
    phoneNumber: '123456789',
    flatId: 2,
    owner: false,
  },
];

export default mockResidentList;
