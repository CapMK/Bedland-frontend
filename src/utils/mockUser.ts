import { User } from 'store/reducers/user/user_slice';

export const mockUserCredentials = {
  residentUsername: 'testmember',
  managerUsername: 'testmanager',
  password: '!Test123!',
};

export const loggedInManagerUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.managerUsername,
  token: undefined,
  role: 'manager',
} as User;

export const loggedInResidentUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.residentUsername,
  token: undefined,
  role: 'resident',
} as User;

export const loggedOutManagerUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'manager',
} as User;

export const loggedOutResidentUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'resident',
} as User;
