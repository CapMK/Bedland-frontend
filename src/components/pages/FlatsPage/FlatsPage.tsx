import Page from 'components/templates/Page/Page';
// eslint-disable-next-line import/no-named-default
import { default as FlatsList } from 'components/organisms/DisplayList/DisplayList';
import SearchBar from 'components/molecules/SearchBar/SearchBar';
import mockResidentFlats from 'utils/mock/mockResidentsFlats';
import * as Styled from './FlatsPage.styled';

const FlatsPage = () => (
  <Page title='Flats'>
    <Styled.FlatsPageContent>
      <Styled.SearchInputContainer>
        <SearchBar placeholder='Search flats...' />
      </Styled.SearchInputContainer>
      <FlatsList displayList={mockResidentFlats} />
    </Styled.FlatsPageContent>
  </Page>
);

export default FlatsPage;
