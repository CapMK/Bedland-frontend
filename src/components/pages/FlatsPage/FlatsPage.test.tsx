import { renderWithProviders } from 'utils/test.utils';
import FlatsPage from './FlatsPage';

describe('FlatsPage', () => {
  it('renders correctly', () => {
    const page = renderWithProviders(<FlatsPage />, {});
    expect(page).toMatchSnapshot();
  });
});
