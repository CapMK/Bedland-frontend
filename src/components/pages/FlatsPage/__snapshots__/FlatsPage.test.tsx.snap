// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FlatsPage > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem;
  position: relative;
  width: 100%;
  z-index: 1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
}

.c13 {
  margin-top: 1rem;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c14:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c10 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c12 {
  padding: 0.5rem 1.031rem 0.938rem;
}

.c9 {
  display: grid;
  gap: 2rem;
  grid-template-columns: repeat(auto-fit,minmax(20rem,1fr));
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-bottom: 1px solid transparent;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.375rem 0rem 0.375rem 0.875rem;
  width: 17.5rem;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  gap: 0.5625rem;
  height: 1.75rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c7 {
  font-family: Source Sans Pro,sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  width: 100%;
}

.c8 {
  position: relative;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin: 0.5rem 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <h2
            style="font-family: Poppins, sans-serif; font-weight: 700; font-size: 1.5rem; line-height: 1.4; color: rgb(27, 26, 26);"
          >
            Flats
          </h2>
        </div>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <div
              class="c4"
            >
              <div>
                <div
                  class="c5"
                >
                  <div
                    class="c6"
                  >
                    <svg
                      data-testid="search-icon"
                      fill="none"
                      height="20"
                      viewBox="0 0 16 16"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <mask
                        id="mask-search"
                      />
                      <path
                        d="M14.6666 14.6666L11.2916 11.2916M11.2916 11.2916C11.8333 10.7499 12.263 10.1068 12.5562 9.39907C12.8494 8.6913 13.0003 7.93272 13.0003 7.16664C13.0003 6.40055 12.8494 5.64197 12.5562 4.9342C12.263 4.22644 11.8333 3.58334 11.2916 3.04164C10.7499 2.49994 10.1068 2.07023 9.39907 1.77707C8.6913 1.4839 7.93272 1.33301 7.16664 1.33301C6.40055 1.33301 5.64197 1.4839 4.9342 1.77707C4.22644 2.07023 3.58334 2.49994 3.04164 3.04164C1.94762 4.13566 1.33301 5.61946 1.33301 7.16664C1.33301 8.71381 1.94762 10.1976 3.04164 11.2916C4.13566 12.3857 5.61946 13.0003 7.16664 13.0003C8.71381 13.0003 10.1976 12.3857 11.2916 11.2916Z"
                        stroke="#4B4B4B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.15972"
                      />
                    </svg>
                    <input
                      class="c7"
                      data-testid="SearchBarInput"
                      placeholder="Search flats..."
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="c8"
                />
              </div>
            </div>
            <div
              aria-labelledby="displaylist"
              class="c9"
              role="list"
            >
              <div
                class="c10"
                role="listitem"
              >
                <img
                  alt="flat"
                  class="c11"
                  src="/src/assets/img/BuildingImg.png"
                />
                <div
                  class="c12"
                >
                  <h3
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                  >
                    Flat #1
                  </h3>
                  <p
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                  >
                    ul. Legnicka 48H, 54-202 Wrocław
                  </p>
                  <div
                    class="c13"
                  >
                    <a
                      class=""
                      href="/buildings/1"
                    >
                      <button
                        class="c14"
                      >
                        Open details
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="c10"
                role="listitem"
              >
                <img
                  alt="flat"
                  class="c11"
                  src="/src/assets/img/BuildingImg.png"
                />
                <div
                  class="c12"
                >
                  <h3
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                  >
                    Flat #2
                  </h3>
                  <p
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                  >
                    ul. Legnicka 48H, 54-202 Wrocław
                  </p>
                  <div
                    class="c13"
                  >
                    <a
                      class=""
                      href="/buildings/2"
                    >
                      <button
                        class="c14"
                      >
                        Open details
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="c10"
                role="listitem"
              >
                <img
                  alt="flat"
                  class="c11"
                  src="/src/assets/img/BuildingImg.png"
                />
                <div
                  class="c12"
                >
                  <h3
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                  >
                    Flat #3
                  </h3>
                  <p
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                  >
                    ul. Legnicka 48H, 54-202 Wrocław
                  </p>
                  <div
                    class="c13"
                  >
                    <a
                      class=""
                      href="/buildings/3"
                    >
                      <button
                        class="c14"
                      >
                        Open details
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="c10"
                role="listitem"
              >
                <img
                  alt="flat"
                  class="c11"
                  src="/src/assets/img/BuildingImg.png"
                />
                <div
                  class="c12"
                >
                  <h3
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                  >
                    Flat #4
                  </h3>
                  <p
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                  >
                    ul. Legnicka 48H, 54-202 Wrocław
                  </p>
                  <div
                    class="c13"
                  >
                    <a
                      class=""
                      href="/buildings/4"
                    >
                      <button
                        class="c14"
                      >
                        Open details
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="c10"
                role="listitem"
              >
                <img
                  alt="flat"
                  class="c11"
                  src="/src/assets/img/BuildingImg.png"
                />
                <div
                  class="c12"
                >
                  <h3
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                  >
                    Flat #1
                  </h3>
                  <p
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                  >
                    ul. Kolorowa 6, 60-198 Poznań
                  </p>
                  <div
                    class="c13"
                  >
                    <a
                      class=""
                      href="/buildings/5"
                    >
                      <button
                        class="c14"
                      >
                        Open details
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem;
  position: relative;
  width: 100%;
  z-index: 1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
}

.c13 {
  margin-top: 1rem;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c14:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c10 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c12 {
  padding: 0.5rem 1.031rem 0.938rem;
}

.c9 {
  display: grid;
  gap: 2rem;
  grid-template-columns: repeat(auto-fit,minmax(20rem,1fr));
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-bottom: 1px solid transparent;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.375rem 0rem 0.375rem 0.875rem;
  width: 17.5rem;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  gap: 0.5625rem;
  height: 1.75rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c7 {
  font-family: Source Sans Pro,sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  width: 100%;
}

.c8 {
  position: relative;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin: 0.5rem 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <h2
          style="font-family: Poppins, sans-serif; font-weight: 700; font-size: 1.5rem; line-height: 1.4; color: rgb(27, 26, 26);"
        >
          Flats
        </h2>
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <div>
              <div
                class="c5"
              >
                <div
                  class="c6"
                >
                  <svg
                    data-testid="search-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 16 16"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <mask
                      id="mask-search"
                    />
                    <path
                      d="M14.6666 14.6666L11.2916 11.2916M11.2916 11.2916C11.8333 10.7499 12.263 10.1068 12.5562 9.39907C12.8494 8.6913 13.0003 7.93272 13.0003 7.16664C13.0003 6.40055 12.8494 5.64197 12.5562 4.9342C12.263 4.22644 11.8333 3.58334 11.2916 3.04164C10.7499 2.49994 10.1068 2.07023 9.39907 1.77707C8.6913 1.4839 7.93272 1.33301 7.16664 1.33301C6.40055 1.33301 5.64197 1.4839 4.9342 1.77707C4.22644 2.07023 3.58334 2.49994 3.04164 3.04164C1.94762 4.13566 1.33301 5.61946 1.33301 7.16664C1.33301 8.71381 1.94762 10.1976 3.04164 11.2916C4.13566 12.3857 5.61946 13.0003 7.16664 13.0003C8.71381 13.0003 10.1976 12.3857 11.2916 11.2916Z"
                      stroke="#4B4B4B"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.15972"
                    />
                  </svg>
                  <input
                    class="c7"
                    data-testid="SearchBarInput"
                    placeholder="Search flats..."
                    value=""
                  />
                </div>
              </div>
              <div
                class="c8"
              />
            </div>
          </div>
          <div
            aria-labelledby="displaylist"
            class="c9"
            role="list"
          >
            <div
              class="c10"
              role="listitem"
            >
              <img
                alt="flat"
                class="c11"
                src="/src/assets/img/BuildingImg.png"
              />
              <div
                class="c12"
              >
                <h3
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                >
                  Flat #1
                </h3>
                <p
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  ul. Legnicka 48H, 54-202 Wrocław
                </p>
                <div
                  class="c13"
                >
                  <a
                    class=""
                    href="/buildings/1"
                  >
                    <button
                      class="c14"
                    >
                      Open details
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="c10"
              role="listitem"
            >
              <img
                alt="flat"
                class="c11"
                src="/src/assets/img/BuildingImg.png"
              />
              <div
                class="c12"
              >
                <h3
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                >
                  Flat #2
                </h3>
                <p
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  ul. Legnicka 48H, 54-202 Wrocław
                </p>
                <div
                  class="c13"
                >
                  <a
                    class=""
                    href="/buildings/2"
                  >
                    <button
                      class="c14"
                    >
                      Open details
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="c10"
              role="listitem"
            >
              <img
                alt="flat"
                class="c11"
                src="/src/assets/img/BuildingImg.png"
              />
              <div
                class="c12"
              >
                <h3
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                >
                  Flat #3
                </h3>
                <p
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  ul. Legnicka 48H, 54-202 Wrocław
                </p>
                <div
                  class="c13"
                >
                  <a
                    class=""
                    href="/buildings/3"
                  >
                    <button
                      class="c14"
                    >
                      Open details
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="c10"
              role="listitem"
            >
              <img
                alt="flat"
                class="c11"
                src="/src/assets/img/BuildingImg.png"
              />
              <div
                class="c12"
              >
                <h3
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                >
                  Flat #4
                </h3>
                <p
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  ul. Legnicka 48H, 54-202 Wrocław
                </p>
                <div
                  class="c13"
                >
                  <a
                    class=""
                    href="/buildings/4"
                  >
                    <button
                      class="c14"
                    >
                      Open details
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="c10"
              role="listitem"
            >
              <img
                alt="flat"
                class="c11"
                src="/src/assets/img/BuildingImg.png"
              />
              <div
                class="c12"
              >
                <h3
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                >
                  Flat #1
                </h3>
                <p
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  ul. Kolorowa 6, 60-198 Poznań
                </p>
                <div
                  class="c13"
                >
                  <a
                    class=""
                    href="/buildings/5"
                  >
                    <button
                      class="c14"
                    >
                      Open details
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
